core/src/execution/mod.rs

```diff
@@ -360,6 +398,8 @@ impl<N: NetworkSpec, R: ExecutionRpc<N>> ExecutionClient<N, R> {
     }
 
     pub async fn uninstall_filter(&self, filter_id: U256) -> Result<bool> {
+        // remove the filter from the state
+        self.state.remove_filter(&filter_id).await;
```

Comment:
What if it is a log filter? Wouldn't we need to uninstall the filter in the rpc if it is a log filter?

Reply:
Yes, that's what the next line (L403) does. Regardless of log type, we always call both `state.remove_filter` and `rpc.uninstall_filter`.

https://github.com/eshaan7/helios/blob/d01f43245c87e07222cde003cd8f57348b8dad8e/core/src/execution/mod.rs#L403